public/app/features/annotations/event_editor.ts

```diff
@@ -48,39 +49,32 @@ export class EventEditorCtrl {
       }
     }
 
+    let crudFunction = saveAnnotation;
     if (saveModel.id) {
-      updateAnnotation(saveModel)
-        .then(() => {
-          this.panelCtrl.refresh();
-          this.close();
-        })
-        .catch(() => {
-          this.panelCtrl.refresh();
-          this.close();
-        });
-    } else {
-      saveAnnotation(saveModel)
-        .then(() => {
-          this.panelCtrl.refresh();
-          this.close();
-        })
-        .catch(() => {
-          this.panelCtrl.refresh();
-          this.close();
-        });
+      crudFunction = updateAnnotation;
+    }
+
+    try {
+      await crudFunction(saveModel);
+    } catch (err) {
+      console.log(err);
+    } finally {
+      this.panelCtrl.refresh();
+      this.close();
+      getDashboardQueryRunner().run({ dashboard: this.panelCtrl.dashboard, range: this.panelCtrl.range });
```

Comment:
I think technically  getDashboardQueryRunner().run({ dashboard: this.panelCtrl.dashboard, range: this.panelCtrl.range });  should be enough here, issuing another metrics query should not be nessary?

Reply:
I mean if we remove `  this.panelCtrl.refresh();` this should still work

Reply:
Ok, so we should remove `this.panelCtrl.refresh();`?

Reply:
Never mind, saw your comment :)